src/pieces.py

```diff
@@ -9,118 +9,272 @@
+        If your king is checked, no other piece can move except for the ones and with only the moves that resolve the check.
+    """
+
+    north: ClassVar[Vector] = Vector(-1, 0)
+    south: ClassVar[Vector] = Vector(+1, 0)
+
+    west: ClassVar[Vector] = Vector(0, -1)
+    east: ClassVar[Vector] = Vector(0, +1)
+
+#   Straight steps:
+    rizontals: ClassVar[set[Vector]] = set((
+        west,
+        east,
+    ))
+    verticals: ClassVar[set[Vector]] = set((
+        north,
+        south,
+    ))
+    straights: ClassVar[set[Vector]] = rizontals.union(verticals)
+
+#   Diagonal steps:
+    diagonals: ClassVar[set[Vector]] = set((rizontal + vertical for rizontal, vertical in product(
+        rizontals,
+        verticals,
+    )))
+
+    steps: ClassVar[set[Vector]] = set()  # a set to be filled by various subclasses
+
+    color: Color
+
+    @property
+    def is_black(self):
+        """Binary switch for color, since it is either black or white.
 
-    color: str
+        Returns:
+            Whether piece is black or not.
+        """
+        return self.color == Color.black
 
-    def valid_steps(self):
-        """Give a list of the valid steps for the piece.
+    def __repr__(self) -> str:
+        f"""Represent a {self.__class__.__name__}.
 
-        Raises:
-            NotImplementedError: Needs implementation.
+        Returns:
+            The representation of a {self.__class__.__name__}.
         """
-        raise NotImplementedError
+        return " "  # An unspecified piece is a ghost piece.
 
-    @property
-    def is_black(self) -> bool:
-        """Return whether the piece is a black piece or not.
+#   NOTE: Remember that target resolution is still unresolved.
+    def legal_moves(self, square: Square, condition: Callable[[Square], bool]) -> set[Square]:
+        f"""Generate all legal moves a {self.__class__.__name__} can apriori make.
+
+        Args:
+            square: Square on which piece is placed.
+                Default is no square, in which case displacements are not resolved into squares and generators not unfold.
 
         Returns:
-            Whether the piece is a black piece or not.
+            A list of all possible moves.
         """
-        return self.color == "black"
+        return set()  # A ghost piece cannot move.
 
 
-@dataclass
 class Pawn(Piece):
-    """A pawn."""
+    """A pawn.
+
+    Moves forward a square, unless its first move where it can move two and unless blocked by any piece.
+    It captured forward-diagonally. I can capture en-passant:
+        If an enemy pawn passes yours trying to escape capture, well... it cannot, not for this round.
+    NOTE: That probably means that board will invoke this method multiple times (unti a better way is thought).
+    Upon reaching the end of the board, it is promoted to an officer:
+        - Rook
+        - Bishop
+        - Knight
+        - Queen
+    """
+
+    pawn_step = Piece.south
+
+    steps = Piece.steps.union({
```

Comment:
Since `Piece.steps` is an empty set, it can be confusing creating a union with an empty step. Also, no need for `{` and `}`. Instead try this:
```py
 steps = pawn_step | (pawn_step + Piece.west) | (pawn_step + Piece.east)
```
Feel free to add comments on what each one is.

Reply:
The logic with using a union was to build on the original `ClassVar` of `Piece`. This may be alternatively resolved by using `update`. See [this comment](https://github.com/TrentHawkins/Chess/pull/15#discussion_r958966217).